src/libsync/configfile.cpp

```diff
@@ -647,7 +648,23 @@ void ConfigFile::setProxyType(int proxyType,
         settings.setValue(QLatin1String(proxyPortC), port);
         settings.setValue(QLatin1String(proxyNeedsAuthC), needsAuth);
         settings.setValue(QLatin1String(proxyUserC), user);
-        settings.setValue(QLatin1String(proxyPassC), pass.toUtf8().toBase64());
+
+        if (pass.isEmpty()) {
+            settings.remove(QLatin1String(proxyPassC));
+
+            auto *job = new QKeychain::DeletePasswordJob(Theme::instance()->appName());
```

Comment:
I'd always give a parent to jobs, to avoid them leaking (I know this one auto deletes but still).

Reply:
Not sure how to proceed here because `ConfigFile` is not derived from `QObject` and creating a parent from the stack seems ugly; Just to be sure, I traced the auto deletes and they equal the count of allocations :)

Reply:
Fair enough.